modules/scalacheck/test/src-ce3/weaver/scalacheck/CheckersConcurrencyTest.scala

```diff
@@ -0,0 +1,98 @@
+package weaver
+package scalacheck
+
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.concurrent.duration._
+
+import cats.effect.Outcome._
+import cats.effect.testkit.TestControl
+import cats.effect.{ IO, Ref }
+import cats.syntax.all._
+
+import weaver.TestStatus
+
+import org.scalacheck.Gen
+
+object CheckersConcurrencyTest extends SimpleIOSuite {
+  test("tests should wait for slower tests to succeed before completion") {
+
+    val minTests = 10
+    object CheckersConcurrencyTestNested extends SimpleIOSuite with Checkers {
+
+      override def checkConfig: CheckConfig =
+        super.checkConfig.copy(perPropertyParallelism = minTests * 5,
+                               minimumSuccessful = minTests,
+                               maximumDiscardRatio = 10)
+
+      private val atomicInt = new AtomicInteger(0)
```

Comment:
I'd use a `Ref` in the `loggedTest` closure. Or at the very least move this atomicInt within the closure (as opposed for it being a field in a static object)

Reply:
My first reaction was to use a ref, but how would you execute it? Calling runUnsafe within the Gen.delay(Gen.const(...))  would be messy and is not really functional in the sense that the Generators are still side-effecting. If you can see another way, I am keen. However, using a functional construct, Ref, only to call it in a non-functional way seems messier then just using a clearly side-effecting construct.

Reply:
hah yeah, brainfart on my side. Let's move the atomicInt as part of the `loggedTest` block. Having a mutable var inside an `object` makes me uneasy 😅 , even if the object in question is nested under a lambda.

Reply:
Moving the AtomicInteger into the closure was a good idea :+1:  I have committed that change